src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs

```diff
@@ -13,6 +14,264 @@ namespace Microsoft.Extensions.Logging
     /// </summary>
     internal static class LoggingExtensions
     {
+        private static Action<ILogger, Guid, DateTimeOffset, Exception> _usingFallbackKeyWithExpirationAsDefaultKey;
```

Comment:
Unrelated, we should use a generator (T4?) to generate these classes. It's a pain to maintain them